### PR TITLE
update makefile to add test, build and images targets.

### PR DESCRIPTION
**Pull request by morvencao**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

/sig ibmcloud

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by gyliu513** [comment]
/uncc @roberthbailey
/cc @jichenjc 

@morvencao please squash to one commit.

File: Makefile [hunk 1]
```diff
@@ -1,95 +1,155 @@
...
+	kubectl apply -f config/crds
+
+# Deploy controller in the configured Kubernetes cluster in ~/.kube/config
+deploy: generate_yaml_test
+	cat provider-components.yaml | kubectl apply -f -
+
+############################################################
+# images section
+############################################################
+images: build docker-build
```
Comment (gyliu513) [inline Makefile]: Why `images` need `build`?

Can we rename `docker-build` to `build-images` directly and use it in test-infra?

s/docker-push/push-images/
s/quick-image/build-push-images/

Comment (morvencao) [reply Makefile]: Oh, I see, we use docker multi-stage build in Dockerfile, https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/blob/master/cmd/manager/Dockerfile#L17

So image building will not depend on external binary building, Ok to remove `build` dependency from `images`

**Comment by gyliu513** [comment]
Please delete https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/blob/master/.travis.yml as well.

**Comment by morvencao** [comment]
Done.

**Pull request merged**